selfdrive/test/process_replay/inject_model.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+import os
+import time
+
+
+from tqdm import tqdm
+from cereal.messaging import PubMaster, recv_one, sub_sock
+from tools.lib.framereader import FrameReader
+import subprocess
+
+
+def rreplace(s, old, new, occurrence):
+  li = s.rsplit(old, occurrence)
+  return new.join(li)
+
+
+def regen_model(msgs, pm, frame_reader, model_sock):
+  # Send some livecalibration messages to initalize visiond
+  for msg in msgs:
+    if msg.which() == 'liveCalibration':
+      pm.send('liveCalibration', msg.as_builder())
+
+
+  out_msgs = []
+  fidx = 0
+  for msg in tqdm(msgs):
+    w = msg.which()
+
+
+    if w == 'frame':
+      msg = msg.as_builder()
+
+      img = frame_reader.get(fidx, pix_fmt="rgb24")[0][:,::-1]
+
+      msg.frame.image = img.flatten().tobytes()
+
+      pm.send(w, msg)
+      model = recv_one(model_sock)
+      fidx += 1
+      out_msgs.append(model)
+    elif w == 'liveCalibration':
+      pm.send(w, msg.as_builder())
+
+  return out_msgs
+
+
+def inject_model(msgs, segment_name):
+  if segment_name.count('--') == 2:
+    segment_name = rreplace(segment_name, '--', '/', 1)
+  frame_reader = FrameReader('cd:/'+segment_name.replace("|", "/") + "/fcamera.hevc")
+  import selfdrive.camerad
```

Comment:
Use `manager.prepare_managed_process('camerad')` and `manager.start_managed_process('camerad')`.